srml/treasury/src/lib.rs

```diff
@@ -291,8 +293,12 @@ impl<T: Trait> OnDilution<BalanceOf<T>> for Module<T> {
 		// pre dilution and post-dilution.
 		if !minted.is_zero() && !portion.is_zero() {
 			let total_issuance = T::Currency::total_issuance();
-			let funding = (total_issuance - portion) / portion * minted;
-			<Pot<T>>::mutate(|x| *x += funding);
+			if let Some(funding) = total_issuance.checked_sub(&portion) {
+				let funding = funding / portion;
+				if let Some(funding) = funding.checked_mul(&minted) {
```

Comment:
this looks problematic though. it would be worth writing some tests with high `portion`s (33%, 50%, 66% of total_issuance) and lowish `minted`s to demonstrate the quantisation effect.

we'll want to switch this code to handle this properly soon.